src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.RoslynWorkspace/RazorWorkspaceListener.cs

```diff
@@ -150,30 +118,43 @@ private void EnqueueUpdate(Project? project)
         }
 
         var projectId = project.Id;
-        if (_workQueues.TryGetValue(projectId, out var scheduler))
+        _workQueue.AddWork(projectId);
+    }
+
+    private async ValueTask UpdateCurrentProjectsAsync(ImmutableArray<ProjectId> projectIds, CancellationToken cancellationToken)
+    {
+        if (_workspace is null)
         {
-            _logger.LogTrace("{projectId} scheduling task due to workspace event", projectId);
+            _logger?.LogTrace("No workspace available when trying to write project info");
+            return;
+        }
```

Comment:
Looking through this class, I don't see how we could ever get here if `_workspace` is null. Consider changing this to the following if you like living on the wild side 😁:

```C#
var workspace = _workspace.AssumeNotNull();
```

Then, use the variable below rather than accessing the field again.